### ISSUE TITLE
How to extract the features from resnet101 based on frcnn

### ISSUE THREAD

**Issue by paulcx** [endernewton/tf-faster-rcnn#93]
That should be conv5 if we want extract features from vgg16 and use them for example xgboost training. What should I cope with the resnet101 then?

**Comment by endernewton** [comment]
you can check the function extract_head() at network.py

**Comment by paulcx** [comment]
Thanks @endernewton. Any idea of how to add another layer for example softmax layer after the fc7 ? So far, I can only extract the ? * 2048 feature vectors from fc7 and then train a separate network for the classification. This also means the frcnn can complete two plus one three tasks in one network architecture (classification+bbox regression and use features from bboxes to classification). Then, it may also require to build a new imdb?

**Comment by endernewton** [comment]
it should be easy to do if you to delve into it, yes you need need to modify imdb if you want to add image-level labels

**Comment by paulcx** [comment]
@endernewton Another question may be based on geometric features such as spatially invariant. Have you come cross the Spatial Transformer Networks. It would be interesting if such approach can be used for the bbox regression or even features extraction.

**Comment by endernewton** [comment]
You can try to implement deformable convolutions and see the performance boost as a relevant note

**Comment by paulcx** [comment]
Any hints for how and where I should start with based on the exsiting codes?
